src/app/pages/task-calendar/smart/smart-form/smart-form.component.ts

```diff
@@ -102,16 +102,19 @@ export class SmartFormComponent {
   protected pk: any;
 
   constructor(protected router: Router,
+              protected ws: WebSocketService,
               protected taskService: TaskService,
               protected storageService: StorageService,
               protected entityFormService: EntityFormService,
               protected aroute: ActivatedRoute) {
     this.disk_field = _.find(this.fieldSets[0].config, { 'name': 'disks' });
-    this.storageService.listDisks().subscribe((res) => {
-      for (const i in res) {
-        this.disk_field.options.push({ label: res[i].name, value: res[i].identifier })
+    this.ws.call('smart.test.disk_choices').subscribe(
```

Comment:
I believe we also need error handling here?

Reply:
Yes. Fixed this as well.